src/tests/badge.rs

```diff
@@ -25,14 +25,35 @@ struct BadgeRef {
     maintenance_attributes: HashMap<String, String>,
 }
 
-fn set_up() -> (DieselConnection, Crate, BadgeRef) {
-    let (_b, app, _middle) = app();
+struct BadgeTestCrate {
+    app: TestApp,
+    krate: Crate,
+}
 
-    let krate = {
-        let conn = app.diesel_database.get().unwrap();
-        let u = new_user("foo").create_or_update(&conn).unwrap();
-        CrateBuilder::new("badged_crate", u.id).expect_build(&conn)
-    };
+impl BadgeTestCrate {
```

Comment:
Neat way of implementing test helper functions! I like it! Abstracts away the non-test-related details but has them close by if you need them.